### PR TITLE
1676 add attachment support for interactionsession

### PR DESCRIPTION
**Pull request by tsearle**
## Changes

What types of changes does your code introduce?  _Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build related changes
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Other (please describe below)

### CODE REVIEW

File: packages/server/transcription-api/service/file_store_api.py
Comment (xvasi) [inline packages/server/transcription-api/service/file_store_api.py:1]: This should be a client class? change name to FileStoreClient?

Comment (tsearle) [reply packages/server/transcription-api/service/file_store_api.py:1]: I'll rename it to file_store_client (python uses snake case)

File: packages/server/comms-api/routes/vcon.go [hunk 1]
```diff
@@ -190,6 +190,26 @@ func submitAnalysis(sessionId string, req model.VCon, cosService s.CustomerOSSer
 	return ids, nil
 }
 
+func submitAttachments(sessionId string, req model.VCon, cosService s.CustomerOSService, c *gin.Context) ([]string, error) {
+	user := getUser(c, &req)
+
+	var ids []string
+	for _, a := range req.Attachments {
```
Comment (xvasi) [inline packages/server/comms-api/routes/vcon.go]: Please don't use letters for variable names ...

File: packages/server/comms-api/routes/vcon.go [hunk 2]
```diff
@@ -284,6 +304,16 @@ func addVconRoutes(conf *c.Config, rg *gin.RouterGroup, cosService s.CustomerOSS
 			ids = append(ids, newIds...)
 		}
 
+		if req.Attachments != nil && len(req.Attachments) > 0 {
+			_, err = submitAttachments(sessionId, req, cosService, c)
+			if err != nil {
```
Comment (xvasi) [inline packages/server/comms-api/routes/vcon.go:309]: who is C?

File: packages/server/comms-api/service/customer_os_service.go [hunk 3]
```diff
@@ -23,6 +23,40 @@ type CustomerOSService interface {
...
+						attachmentId: $attachmentId
+				) {
+						id
+				}
+			}`)
+
+	graphqlRequest.Var("sessionId", sessionId)
+	graphqlRequest.Var("attachmentId", attachmentId)
+
+	err := s.addHeadersToGraphRequest(graphqlRequest, tenant, user)
```
Comment (xvasi) [inline packages/server/comms-api/service/customer_os_service.go]: s?

File: packages/server/file-store-api/main.go [hunk 4]
```diff
@@ -77,13 +78,13 @@ func main() {
 
 			multipartFileHeader, err := c.FormFile("file")
 			if err != nil {
-				c.AbortWithStatus(500) //todo
```
Comment (xvasi) [inline packages/server/file-store-api/main.go:80]: c ?

File: packages/server/transcription-api/routes/transcription_service.py [hunk 5]
```diff
@@ -130,9 +144,10 @@ def handle_transcribe_post_request():
     print("Parties: " + str(parties))
 
     vcon_api = VConPublisher(os.environ.get('VCON_API_URL'), os.environ.get('VCON_API_KEY'), request.headers.get('X-Openline-USERNAME'), parties)
+    fs_api = file_store_api.FileStoreApi(os.environ.get('FILE_STORE_API_URL'), os.environ.get('FILE_STORE_API_KEY'), request.headers.get('X-Openline-USERNAME'))
 
     # Start a new thread to process the file
-    t = threading.Thread(target=process_file, args=(file_to_process, participants, topic, vcon_api))
```
Comment (xvasi) [inline packages/server/transcription-api/routes/transcription_service.py:135]: t -> thread

File: packages/server/transcription-api/service/file_store_api.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import requests
+
+class FileStoreApi:
```
Comment (xvasi) [inline packages/server/transcription-api/service/file_store_api.py]: This is an actuall Client not the API

File: packages/server/transcription-api/transcribe/transcribe.py [hunk 7]
```diff
@@ -189,33 +192,48 @@ def build_transcribe_prompt(participants, industries, descriptions, topic):
...
-        input={"audio": buffer, "initial_prompt": prompt,
-               "condition_on_previous_text": True,
-               "compression_ratio_threshold": 2.4,
-               "logprob_threshold": -1,
-               "temperature": temperature, }
-    )
+def transcribe_audio_buffer(audio_file, prompt, temperature):
+    with open(audio_file, "rb") as f:
+        segment_output = replicate.run(
+            "openai/whisper:e39e354773466b955265e969568deb7da217804d8e771ea8c9cd0cef6591f8bc",
```
Comment (xvasi) [inline packages/server/transcription-api/transcribe/transcribe.py]: can you extract this into an env var or at least a variable?

**Review by xvasi — CHANGES_REQUESTED** [review]

File: packages/server/file-store-api/main.go (hunk 4 above)
Comment (tsearle) [reply packages/server/file-store-api/main.go:80]: c is the context, it is pretty much standard in our entire codebase

File: packages/apps/spaces/components/ui-kit/atoms/message/TranscriptContent.tsx [hunk 8]
```diff
@@ -30,57 +28,15 @@ export const TranscriptContent: React.FC<TranscriptContentProps> = ({
...
-
         return (
-          <div
-            key={index}
-            className={classNames(styles.singleMessage, {
-              [styles.isleft]: transcriptElement?.party.tel,
-              [styles.isright]: !transcriptElement?.party.tel,
-            })}
+          <Message
+            transcriptElement={transcriptElement}
```
Comment (Kasia9090) [inline packages/apps/spaces/components/ui-kit/atoms/message/TranscriptContent.tsx:32]: Please add `key` attribute

**Review by Kasia9090 — CHANGES_REQUESTED** [review]

**Review by Kasia9090 — DISMISSED** [review]

**Review by xvasi — DISMISSED** [review]

**Review by Kasia9090 — APPROVED** [review]

**Pull request merged**